### ISSUE TITLE
Have a version with PHP 7.2 support

### ISSUE THREAD

**Issue by oliverlundquist** [neomerx/json-api#189]
Hi, thanks for a great library.

I stumbled upon this error when upgrading to PHP 7.2 today.

This: https://github.com/neomerx/json-api/blob/2216645b3a5a6851907d85686506342397447f78/src/Schema/Container.php#L163

Gives: 
```
Error: Failure evaluating code:
is_callable($classNameOrCallable) === true
ErrorException: assert(): Calling assert() with a string argument is deprecated
```

I guess it would be enough just changing this like to:
```
assert(is_callable($classNameOrCallable));
```

Thanks

**Comment by neomerx** [comment]
Yep, that's PHP incompatibility. PHP 5.6 accepts only strings, 7.0 and 7.1 both and 7.2 only non-strings. Unfortunately, we can't have both.
It's a broader issue than just removing `'` from asserts. PHP 7+ brings to table stronger type system and what should be done is preparing version 2.0 for PHP 7.2+.

**Comment by neomerx** [comment]
2 step solution.

Step 1. Make the lib compatible to 7.2 without losing compatibility with earlier versions.
- done with removing the line with assert. To this date, we can be sure it works fine.

Step 2. Make a new version taking advantages of PHP 7.2
- in the process. A branch called `v2` is created.

**Comment by neomerx** [comment]
Step 1 released as [1.0.8](https://github.com/neomerx/json-api/releases/tag/v1.0.8)

For step 2 issue #190 is added
